pkg/middleware/recovery/recovery.go

```diff
@@ -0,0 +1,48 @@
+package recovery
+
+import (
+	"context"
+	"net/http"
+
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+func Panic(next http.Handler, logger logger.Logger) http.Handler {
```

Comment:
We should have 2 of these, one for the HTTP mux and one for the gRPC server:

1. `HTTPPanicRecoveryHandler(next http.Handler, logger ...) http.handler`

2. `PanicRecoveryHandler(logger Logger) recovery.RecoveryHandlerFuncContext`

The grpc `PanicRecoveryHandler` should be used in `run.go` like so:
```go
grpc_recovery.UnaryServerInterceptor(
	grpc_recovery.WithRecoveryHandlerContext(
        recovery.PanicRecoveryHandler(logger),
    ),
),
```

Reply:
Done